docker/release-images.mjs

```diff
@@ -230,6 +299,28 @@ function getImageReleaseGroupByImageTagPrefix(prefix) {
   return "aptos-node";
 }
 
+const APTOS_RELEASE_REGEX = /aptos-node-v(\d+\.\d+\.\d+)/;
+
+export function assertTagMatchesSourceVersion(imageTag) {
+  const config = toml.parse(fs.readFileSync("aptos-node/Cargo.toml"));
+  const configVersion = config.package.version;
+  if (!doesTagMatchConfig(imageTag, configVersion)) {
+    reportError(`image tag does not match cargo version: ${imageTag} !== ${configVersion}`, {throwOnFailure: true});
+  }
+}
+
+export function doesTagMatchConfig(imageTag, configVersion) {
+  if (!APTOS_RELEASE_REGEX.test(imageTag)) {
```

Comment:
`if (!isReleaseImage(imageTag))` ?

Reply:
Actually here we should rely on the regex since we match in it below, this ensures the match below will work so we actually do not want to see if its a realase image their implementation is just currently the same